matrix-sdk/src/main/java/org/matrix/androidsdk/data/cryptostore/MXFileCryptoStore.java

```diff
@@ -381,14 +392,21 @@ private boolean storeObject(Object object, File file, String description) {
                 }
 
                 FileOutputStream fos = new FileOutputStream(file);
-                GZIPOutputStream gz = CompatUtil.createGzipOutputStream(fos);
-                ObjectOutputStream out = new ObjectOutputStream(gz);
+                CipherOutputStream cos = CompatUtil.createCipherOutputStream(fos, mContext);
 
-                out.writeObject(object);
-                out.close();
+                if (cos != null) {
+                    GZIPOutputStream gz = CompatUtil.createGzipOutputStream(cos);
+                    ObjectOutputStream out = new ObjectOutputStream(gz);
 
-                succeed = true;
-                Log.d(LOG_TAG, "## storeObject () : " + description + " done in " + (System.currentTimeMillis() - t0) + " ms");
+                    out.writeObject(object);
+                    out.flush();
+                    out.close();
+
+                    succeed = true;
+                    Log.d(LOG_TAG, "## storeObject () : " + description + " done in " + (System.currentTimeMillis() - t0) + " ms");
+                } else {
+                    Log.e(LOG_TAG, "## writeObject() failed to open cipher output stream");
```

Comment:
You should write object unencrypted for compatibility mode

Reply:
Are you sure this is a good idea ?